### PR TITLE
modify plot set 5 and other files for rmse and correlations

### PR DESCRIPTION
**Pull request by mcenerney1**
The branch rmse has the required changes to compute/display the rmse and correlation for plot set 5.

### CODE REVIEW

**Comment by painter1** [comment]
looks good to me.  The only question is whether to squeeze it in now, or wait for the 2.4.1 release.  I'd wait, so we don't need another testing cycle.

File: src/python/packages/amwg/amwg.py [hunk 1]
```diff
@@ -1252,7 +1252,29 @@ class amwg_plot_set5(amwg_plot_set5and6):
...
     name = '5 - Horizontal Contour Plots of Seasonal Means'
     number = '5'
-    
+    def customizeTemplates(self, templates, var=None):
+        """Theis method does what the title says.  It is a hack that will no doubt change as diags changes."""
+        (cnvs1, tm1), (cnvs2, tm2) = templates
+        import pdb
+        if hasattr(var, 'RMSE'):
+            RMSE = round(var.RMSE, 2)
+            CORR = round(var.CORR, 2)
```
Comment (doutriaux1) [inline src/python/packages/amwg/amwg.py]: you probably want an `and hasattr(var,'CORR')` in there or even better split this in two sections one for `RMSE` one for `CORR`. No?

Comment (mcenerney1) [inline src/python/packages/amwg/amwg.py]: Clearly this is not perfect. I should put a try block where they get computed also. Once Jonathas' branch is merged we'll need to revisit this.

**Comment by painter1** [comment]
Note that RMSE and CORR are both averages, so some attention should be paid to weighting.  In particular, some time ago Susannah Burrows had requested mass weighting for RMSE.   I'm saying this just for information; the weighting issue shouldn't prevent a merge of this pull request.  I'll put something about this in the issue tracker.

**Comment by doutriaux1** [comment]
@painter1 you're right these are area-weighted not mass weighted, definitely should be added here or later.

File: src/python/computation/reductions.py [hunk 2]
```diff
@@ -2386,6 +2400,17 @@ def aminusb_2ax( mv1, mv2, axes1=None, axes2=None ):
...
+    
+    #compute rmse and correlations
+    import genutil.statistics, numpy
+    aminusb.RMSE = numpy.infty
+    aminusb.CORR = numpy.infty
+    try:
+        aminusb.RMSE = float( genutil.statistics.rms(mv1new, mv2new, axis='xy') )
+        aminusb.CORR = float( genutil.statistics.correlation(mv1new, mv2new, axis='xy') )
+    except Exception,err:
+        print err, "<<<<<<<<<<<<<<<<<<<"
```
Comment (doutriaux1) [inline src/python/computation/reductions.py]: @mcenerney1 should this be just `pass`

File: src/python/frontend/uvcdat.py [hunk 3]
```diff
@@ -489,9 +489,15 @@ def finalize( self, flip_x=False, flip_y=False ):
                     nlevels = 16
     
                     try:
-                        levels = [float(v) for v in vcs.mkscale( varmin, varmax, nlevels )]
+                        #changed by Charles on 4/15/16 to support Chris
```
Comment (doutriaux1) [inline src/python/frontend/uvcdat.py]: @mcenerney1 do we want that in 2.4.1?

File: src/python/packages/amwg/amwg.py [hunk 4]
```diff
@@ -1252,7 +1252,29 @@ class amwg_plot_set5(amwg_plot_set5and6):
...
+            textRMSE = cnvs2.createtext()
+            textRMSE.string = 'RMSE = %.3g' % RMSE
+            textRMSE.x = .075
+            textRMSE.y = .005
+            textRMSE.height = 10
+            cnvs2.plot(textRMSE, bg=1)
```
Comment (doutriaux1) [inline src/python/packages/amwg/amwg.py]: is it always `bg=1`?

**Pull request merged**